packages/ember-glimmer/tests/integration/helpers/input-test.js

```diff
@@ -0,0 +1,446 @@
+  }
+
+  assertSingleInput() {
+    this.assert.equal(this.$('input').length, 1, 'A single text field was inserted');
+  }
+
+  assertSingleCheckbox() {
+    this.assert.equal(this.$('input[type=checkbox]').length, 1, 'A single checkbox is added');
+  }
+
+  assertCheckboxIsChecked() {
+    this.assert.equal(this.$input().prop('checked'), true, 'the checkbox is checked');
+  }
+
+  assertCheckboxIsNotChecked() {
+    this.assert.equal(this.$input().prop('checked'), false, 'the checkbox is not checked');
+  }
+
+  assertValue(expected) {
+    this.assert.equal(this.$input().val(), expected, `the input value should be ${expected}`);
+  }
+
+  assertAttr(name, expected) {
+    this.assert.equal(this.$input().attr(name), expected, `the input ${name} attribute has the value '${expected}'`);
+  }
+
+  assertAllAttrs(names, expected) {
+    names.forEach((name) => this.assertAttr(name, expected));
+  }
+
+  assertSelectionRange(start, end) {
+    let input = this.$input()[0];
+    this.assert.equal(input.selectionStart, start, `the cursor start position should be ${start}`);
+    this.assert.equal(input.selectionEnd, end, `the cursor end position should be ${end}`);
+  }
+}
+
+moduleFor('@htmlbars Helpers test: {{input}}', class extends InputRenderingTest {
+
+  ['@test a single text field is inserted into the DOM'](assert) {
+    this.render(`{{input type="text" value=value}}`, { value: 'hello' });
+
+    let id = this.inputID();
+
+    this.assertValue('hello');
+    this.assertSingleInput();
+
+    this.runTask(() => this.rerender());
+
+    this.assertValue('hello');
+    this.assertSingleInput();
+    this.assertInputId(id);
+
+    this.runTask(() => set(this.context, 'value', 'goodbye'));
+
+    this.assertValue('goodbye');
+    this.assertSingleInput();
+    this.assertInputId(id);
+
+    this.runTask(() => set(this.context, 'value', 'hello'));
+
+    this.assertValue('hello');
+    this.assertSingleInput();
+    this.assertInputId(id);
+  }
+
+  ['@test default type']() {
+    this.render(`{{input}}`);
+
+    this.assertAttr('type', 'text');
+
+    this.runTask(() => this.rerender());
+
+    this.assertAttr('type', 'text');
+  }
+
+  ['@test dynamic attributes']() {
+    this.render(`
+      {{input type="text"
+        disabled=disabled
+        value=value
+        placeholder=placeholder
+        name=name
+        maxlength=maxlength
+        size=size
+        tabindex=tabindex
+      }}`, {
+        disabled: false,
+        value: 'Original value',
+        placeholder: 'Original placeholder',
+        name: 'original-name',
+        maxlength: 10,
+        size: 20,
+        tabindex: 30
+      }
+    );
+
+    this.assertNotDisabled();
+    this.assertValue('Original value');
+    this.assertAttr('placeholder', 'Original placeholder');
+    this.assertAttr('name', 'original-name');
+    this.assertAttr('maxlength', '10');
+    // this.assertAttr('size', '20');
```

Comment:
@krisselden I was incorrect in my last comment. The `input` helper hasn't yet been implemented in `glimmer`, the commented out lines seem to cover existing bugs in `htmlbars`.